pkg/ecdsa/retry.go

```diff
@@ -0,0 +1,62 @@
+package ecdsa
+
+import (
+	"fmt"
+	"math/rand"
+	"sort"
+
+	"github.com/keep-network/keep-core/pkg/chain"
+)
+
+type byAddress []chain.Address
+
+func (ba byAddress) Len() int           { return len(ba) }
+func (ba byAddress) Swap(i, j int)      { ba[i], ba[j] = ba[j], ba[i] }
+func (ba byAddress) Less(i, j int) bool { return ba[i] < ba[j] }
+
+func EvaluateRetryParticipantsForSigning(
+	groupMembers []chain.Address,
+	seed int64,
+	retryCount uint,
+	quantity uint,
+) ([]chain.Address, error) {
+	if int(quantity) > len(groupMembers) {
+		return nil, fmt.Errorf(
+			"Asked for too many seats. %d seats were requested, but there are only %d available.",
+			quantity,
+			len(groupMembers),
+		)
+	}
+	operatorToSeatCount := make(map[chain.Address]uint)
+	for _, operator := range groupMembers {
+		operatorToSeatCount[operator]++
+	}
+	rand.Seed(seed + int64(retryCount))
+
+	operators := make(byAddress, len(operatorToSeatCount))
+	i := 0
+	for operator := range operatorToSeatCount {
+		operators[i] = operator
+		i++
+	}
+	sort.Sort(operators)
```

Comment:
You can apply the comparator this way:
```suggestion
	sort.Sort(byAddress(operators))
```

Reply:
[c68a994](https://github.com/keep-network/keep-core/pull/3109/commits/c68a994e6559aea4344ca18b1655f86c619dfed3
)